### ISSUE TITLE
.NET file change notifications do not work

### ISSUE THREAD

**Issue by alonsohki** [billziss-gh/winfsp#337]
## Bug Report
Issuing a file change notify does not arrive to the filesystem watchers.

### How to Reproduce
Have a filesystem issuing a notify operation in the following sequence:
```csharp
FileSystemHost.NotifyBegin(uint.MaxValue);
FileSystemHost.Notify(new NotifyInfo[] { new NotifyInfo {
  FileName = @"\path\to\file",
  Filter = NotifyFilter.ChangeLastWrite,
  Action = NotifyAction.Modified,
} });
FileSystemHost.NotifyEnd();
```

Have a watcher listening for these notifications:
```csharp
using System;
using System.IO;

namespace FsWatcher
{
    class Program
    {
        static void Main(string[] args)
        {
            // Create a new FileSystemWatcher and set its properties.
            using (FileSystemWatcher watcher = new FileSystemWatcher())
            {
                watcher.Path = @"C:\mount\point\file\parent\directory";
                // Watch for changes in LastAccess and LastWrite times, and
                // the renaming of files or directories.
                watcher.NotifyFilter = NotifyFilters.LastAccess
                                     | NotifyFilters.LastWrite
                                     | NotifyFilters.FileName
                                     | NotifyFilters.DirectoryName;
                // Add event handlers.
                watcher.Changed += OnChanged;
                watcher.Created += OnChanged;
                watcher.Deleted += OnChanged;
                watcher.Renamed += OnRenamed;
                // Begin watching.
                watcher.EnableRaisingEvents = true;
                // Wait for the user to quit the program.
                Console.WriteLine("Press 'q' to quit the sample.");
                while (Console.Read() != 'q') ;
            }
        }
        // Define the event handlers.
        private static void OnChanged(object source, FileSystemEventArgs e) =>
            // Specify what is done when a file is changed, created, or deleted.
            Console.WriteLine($"File: {e.FullPath} {e.ChangeType}");
        private static void OnRenamed(object source, RenamedEventArgs e) =>
            // Specify what is done when a file is renamed.
            Console.WriteLine($"File: {e.OldFullPath} renamed to {e.FullPath}");
    }
}
```

1. Modify the file using notepad, and notice how the event is received on the test watcher app.
2. Trigger the event through the WinFsp interface, and notice how the watcher app is not aware of this event.

### Behaviors
It should show up in the watcher app.

### Environment
- OS version and build: Windows 10 Pro, Version	10.0.19042 Build 19042
- WinFsp version and build: Self-built WinFsp on commit hash 6421dd9

**Comment by alonsohki** [comment]
Ping @billziss-gh

**Comment by billziss-gh** [comment]
I believe this has now been addressed by commits 6421dd92a9e2aad5b296b22067182b04a27c02c0 and 80067633671a21029e1debb9473d4a8a13cbf0da.

I have added new filesystems [notifyfs](https://github.com/billziss-gh/winfsp/tree/master/tst/notifyfs) and [notifyfs-dotnet](https://github.com/billziss-gh/winfsp/tree/master/tst/notifyfs-dotnet) that demonstrate use of the file change notification mechanism.

I am closing this, but please reopen if the problem persists.
